api/v1alpha1/managedcluster_types.go

```diff
@@ -62,27 +62,25 @@ const (
 
 // ManagedClusterSpec defines the desired state of ManagedCluster
 type ManagedClusterSpec struct {
-	// DryRun specifies whether the template should be applied after validation or only validated.
-	// +optional
-	DryRun bool `json:"dryRun,omitempty"`
-	// Template is a reference to a Template object located in the same namespace.
-	// +kubebuilder:validation:Required
```

Comment:
We should not remove the `Required` flag here to restrict `ManagedClusters` creation with an empty `spec.template` before the request is passed to the webhook. `MinLength` validation is not enough since an empty string bypasses this check.

Reply:
there were no real changes to the generated CRDs, so I assume those are irrelevant

Reply:
https://github.com/Mirantis/hmc/pull/382/files#diff-98472f9e89833cf1be765a0d990eeb34a6ba3523f073cf7f5c315039149cf680R70

Reply:
@Kshatrix it is so, but suddenly the schema-validation validates even the empty values and the requests proceeds further to the admission, which is really odd

Reply:
due to the `admission.Defaulter` implementation, having no `spec.config` leads to further proceeding, and in particular: schema and then the validationwebhook, so in reality, that chang is actually noop

Reply:
Sorry, nevermind. You're right, we can safely remove it. The mutation happens before the schema validation.